docs/dev/api-versioning.rst

```diff
@@ -0,0 +1,19 @@
+API Versioning
+===========================
+
+The API paths are prefixed with an optional version number. If the version number is not specified, the latest version is assumed. For example, if the latest version is 4, ``/api/v4/...`` is the same as ``/api/...``.
+
+To create a new API version, extend the list of versions in the `VersioningConfiguration`. All endpoints with a lower limit will automatically support the new version.
+
+There a three ways to specify what version an endpoint supports:
+    - No versioning annotation: The endpoint supports all versions Artemis supports.
+    - ``@VersionRange``: The endpoint supports all versions in the specified range or limit.
+    - ``@VersionRanges``: The endpoint supports all versions in the specified ranges or limits. This annotation contains a list of ``@VersionRange`` annotations.
```

Comment:
What is the use case for this? We support versions 1-3 but not version 4 but from version 5 onwards we again support this?

Reply:
The change could be reverted because we see that it will conflict with another design decision in the future. However, the app already uses the new endpoint.

Probably unlikely, but I saw no reason to limit the functionality here. We don't have to use the feature as you can directly annotate the endpoint with a @VersionRange annotation.

Reply:
Would there be any downside to simply extend the range in my example to 1-5 or from 1 onwards to infinity?
The reason against `@VersionRanges` would be more and more complicated code that has to be maintained

Reply:
Yes, if you would simply extend the range, you would have to remove the "new/old/reverted" endpoint and suddenly everything using that endpoint would not be working anymore, but suddenly be confronted with a different functionality behind the same endpoint. Exactly the kind of thing we want to avoid with this PR in the first place.

Point 2: That is true. However, I'd argue that the maintenance is pretty low, if any. If there are significant changes in how Spring allows us to handle this situation - we would have to update everything anyway.